### PR TITLE
Correspondence cost based probability grid

### PR DESCRIPTION
**Pull request by kdaun**
- Store probabilities as correspondence costs
- Add conversion functions for correspondence cost values
- Step towards [RFC 0019](https://github.com/googlecartographer/rfcs/blob/master/text/0019-probability-grid-and-submap2d-restructuring.md)

### CODE REVIEW

File: cartographer/mapping/probability_values.cc [hunk 1]
```diff
@@ -21,35 +21,55 @@ namespace mapping {
...
+      result->push_back(SlowValueToBoundedFloat(
+          value, unknown_value, unknown_result, lower_bound, upper_bound));
     }
   }
   return result;
 }
 
+const std::vector<float>* PrecomputeValueToProbability() {
+  return PrecomputeValueToBoundedFloat(kUnknownProbabilityValue,
+                                       kMinProbability, kMinProbability,
```
[the remark this answers is not in the document]
Comment (pifon2a) [reply cartographer/mapping/probability_values.cc]: oh, sorry, i did not notice that there was `unknown_result` as well as `unknown_value`.

File: cartographer/mapping/probability_values.cc [hunk 2]
```diff
@@ -21,35 +21,55 @@ namespace mapping {
...
+    return unknown_result;
   }
-  const float kScale = (kMaxProbability - kMinProbability) / 32766.f;
-  return value * kScale + (kMinProbability - kScale);
+  const float kScale = (upper_bound - lower_bound) / 32766.f;
+  return value * kScale + (lower_bound - kScale);
 }
 
-const std::vector<float>* PrecomputeValueToProbability() {
+const std::vector<float>* PrecomputeValueToBoundedFloat(
```
[the remark this answers is not in the document]
Comment (kdaun) [reply cartographer/mapping/probability_values.cc]: Yes!

**Comment by kdaun** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user kdaun. Merge queue now has a length of 2.

**Pull request merged**